app/models/big_blue_button_conference.rb

```diff
@@ -87,21 +113,90 @@ def participant_join_url(user, return_to = "http://www.instructure.com")
 
   def recordings
     fetch_recordings.map do |recording|
-      recording_format = recording.fetch(:playback, {}).fetch(:format, {})
-      {
-        recording_id:     recording[:recordID],
-        duration_minutes: recording_format[:length].to_i,
-        playback_url:     recording_format[:url],
+      recording_formats = recording.fetch(:playback, {})
+      recordingObj = {
+        recording_id:       recording[:recordID],
+        recording_vendor:   "big_blue_button",
+        published:          recording[:published]=="true" ? true : false,
+        protected:          recording[:protected] ? recording[:protected]=="true" ? true : false : nil,
+        ended_at:           recording[:endTime].to_i,
+        duration_minutes:   recording_formats.first[:length].to_i,
+        recording_formats:  [],
+        images:             []
       }
+      recording_formats.each do |recording_format|
+        recordingObj[:recording_formats] << {
+          type:             recording_format[:type].capitalize,
+          playback_url:     recording_format[:url]
+        }
+        if recording_format[:preview] && recording_format[:preview][:images] && recording_format[:preview][:images].length > recordingObj[:images].length
+          recordingObj[:images] = recording_format[:preview][:images]
+        end
+      end
+      recordingObj
     end
   end
 
   def delete_all_recordings
-    fetch_recordings.map do |recording|
-      delete_recording recording[:recordID]
+    recordings = fetch_recordings.map!{ |recording| recording[:recordID] }
+    if recordings.length > 1 && recordings.length < 10
+      response = send_request(:deleteRecordings, {
+        :recordID => recordings.join(",")
+        })
+      response[:deleted] if response
+    elsif recordings.length == 1 || recordings.length >= 10
+      recordings.each{ |recording_id| delete_recording(recording_id) }
     end
   end
 
+  def delete_recording(recording_id)
+    response = send_request(:deleteRecordings, {
+      :recordID => recording_id,
+      })
+    response[:deleted] if response
+  end
+
+  def publish_recording(recording_id, publish)
+    send_request(:publishRecordings, {
+      :recordID => recording_id,
+      :publish  => publish
+      })
+    response = { :published => "false", :recording_formats => [] }
+    if publish=="true"
+      recording = nil
+      begin
+        response_recordings = send_request(:getRecordings, {
+        :meetingID => conference_key
+        })
+        recording = response_recordings[:recordings].find{ |r| r[:recordID]==recording_id }
+        sleep(10) if recording[:published]=="false"
+      end while recording.nil? || recording[:published]=="false"
```

Comment:
Wow, I didn't see that. I can't understand the intention of that, and I don't have the guy who could explain it to me :). Anyways, it is gone now